packages/telemetry/index.cjs

```diff
@@ -46,40 +77,84 @@ const logEvent = async (eventName, dev, settings) => {
 				repo = md5(settings.gitRepo);
 			}
 
-			if (settings.database) {
+			if (databaseName) {
+				database = databaseName;
+			} else if (settings.database) {
+				//legacy - remove this post migration
 				database = settings.database;
 			}
 
-			if (settings.credentials.filename) {
+			if (settings.credentials?.filename) {
 				demoDb = md5(settings.credentials.filename) === md5('needful_things.duckdb');
 			}
 		}
 
-		let directoryHash = md5(process.env.HOME);
-		let codespaces = process.env.CODESPACES === 'true';
+		let homeDirectory = undefined;
+		let codespaces = false;
+
+		if (process.env) {
+			const { HOME, CODESPACES } = process.env;
+			homeDirectory = HOME;
+			if (CODESPACES) {
+				codespaces = CODESPACES === 'true';
+			}
+		}
 
 		if (usageStats === 'yes') {
-			projectProfile = await getProfile();
+			const projectProfile = await getProfile();
 			var analytics = new Analytics(wK);
-			analytics.track({
+			const payload = {
 				anonymousId: projectProfile.anonymousId,
 				event: eventName,
 				properties: {
 					devMode: dev,
 					repoHash: repo,
 					database: database, // logs database type (postgres, snowflake, etc.)
+					sourceNameHash: sourceName ? md5(sourceName) : undefined, //logs the hashed name of the source this is associated with (e.g md5('pet-store')))
+					queryNameHash: queryName ? md5(queryName) : undefined, //logs the hashed name of the query this is associated with (e.g md5('pet-store')))
 					operatingSystem: process.platform, // logs operating system name
 					nodeVersion: process.version, // logs active version of NodeJS
 					arch: process.arch,
-					directoryHash: directoryHash,
+					directoryHash: homeDirectory ? md5(homeDirectory) : undefined,
 					demoDb: demoDb,
-					codespaces: codespaces
+					codespaces: codespaces,
+					postUSQL: true
 				}
-			});
+			};
+			console.debug('analytics.track', { payload });
 		}
 	} catch {
 		// do nothing
 	}
 };
 
-module.exports = logEvent;
+/**
+ * Logs an event emiited from source queries
+ * @param {TelemetryEventName} eventName
+ * @param {string | undefined} [databaseName]
+ * @param {string | undefined} [sourceName]
+ * @param {string | undefined} [queryName]
+ */
+const logQueryEvent = async (eventName, databaseName, sourceName, queryName, dev = false) => {
+	//TODO there is no concept of dev mode when running npm run sources
```

Comment:
@hughess I don't think we can easily tell apart `npm run sources` commands run by CI vs during development.  What I did for now was to set  `dev=false` for all events emitted as a result of `npm run sources`

@ItsMeBrianD or @csjh any alternative ideas?  I just thought about checking for the existence of evidence.setting.json.

Reply:
@ud3sh doesn't the built-in svelte `dev` variable cover that for us?

Reply:
Not necessarily, since `npm run sources` isn't run while the dev server is running

I think we could check the node_env environment variable to determine source HMR though

Reply:
ahhh got it

Reply:
I don't recall seeing the `NODE_ENV` var when runnign `npm run sources` (I printed these out to see).  It wouldn't make sense for it to be populated I'd think unless you run `NODE_ENV=development npm run dev`.

The evidence.settings.json wouldn't be available in non-dev environment right?
- it's not source controlled hence when you run in on CI, it won't be available.

Moving this back to draft.  I think this is something we can solve.

Reply:
If we're blocked from everything else in the meantime, it might be worth going ahead without the dev mode indicator and adding it later. Not sure what info will be available prior to this PR going in

Reply:
> If we're blocked from everything else in the meantime, it might be worth going ahead without the dev mode indicator and adding it later. Not sure what info will be available prior to this PR going in

No events are going in post USQL.